doc/user-manual/language/with-abstraction.lagda.rst

```diff
@@ -381,6 +381,64 @@ with-abstraction may change the type of variables, the instantiation
 of such let-bound variables are type checked again after
 with-abstraction.
 
+..
+  ::
+  module with-invert {a b} {A : Set a} {B : A → Set b} where
+    open import Agda.Builtin.Sigma using (Σ; _,_)
+
+.. _with-invert:
+
+Invert
+~~~~~~
+
+When a pattern is irrefutable, we can use ``invert`` instead of a ``with``
+block. This gives us a lightweight syntax to make a lot of observations
+before using a "proper" ``with`` block. See for instance the redefinitions
+of the first and second projections out of a record::
+
+    module _ (p : Σ A B) where
+
+      fst : A
+      fst invert (a , _) ← p = a
+
```

Comment:
This could also be done with
```agda
fst = let (a , _) = p in a
```
The real power of `invert` shows for indexed inductive types.
Maybe your example becomes more meaningful when combined with `with`.

While I am writing this, maybe `let` could be used instead of `invert`?